File: scala-server/app/com/bolour/eagerwords/common/domain/GameSummary.scala
/*
 * Copyright 2017-2020 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/eagerwords.com/blob/master/LICENSE.md
 */
package com.bolour.eagerwords.common.domain

/**
  * Summary of game reported to clients after the last play.
  *
  * @param stopInfo Data about why the last play was reached.
  */
case class GameSummary(stopInfo: StopInfo)

File: haskell-server/src/EagerWords/Common/Domain/GameSummary.hs
--
-- Copyright 2017-2020 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/eagerwords/blob/master/LICENSE.md
--

{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module EagerWords.Common.Domain.GameSummary (
  GameSummary(..)
) where

import GHC.Generics (Generic)
import Control.DeepSeq (NFData)
import Data.Aeson (FromJSON, ToJSON)
import EagerWords.Common.Domain.StopInfo (StopInfo)

-- | Summary of the game at the end.
data GameSummary = GameSummary {
    stopInfo :: StopInfo
}
  deriving (Eq, Show, Generic, NFData)
  
instance FromJSON GameSummary
instance ToJSON GameSummary

